src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,36 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+
+  constructor(name) {
+    this.name = name;
+    this.health = 100;
+    Animal.alive.push(this);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(animal) {
+    if (animal instanceof Herbivore && !animal.hidden && animal.health > 0) {
+      animal.health -= 50;
+
+      if (animal.health <= 0) {
+        Animal.alive = Animal.alive.filter((a) => a.health > 0);
+      }
+    }
```

Comment:
[CODE PERFORMANCE] - The `bite` method currently filters the `Animal.alive` array every time an animal's health drops below 0. This can be inefficient if multiple animals are bitten in a short period, as it would repeatedly filter the entire array. Consider checking and filtering the `Animal.alive` array after all bite actions are performed to improve performance, or implement a separate method to handle the removal of dead animals from the array.